capytaine/io/xarray.py

```diff
@@ -300,40 +324,184 @@ def assemble_dataset(results: Sequence[LinearPotentialFlowResult],
+        else:
+            dataset.coords['wavelength'] = 2*np.pi/wavenumber_data_array(results)
 
     if mesh:
-        # TODO: Store full mesh...
-        bodies = list({result.body for result in results})  # Filter out duplicate bodies in the list of results
-        nb_faces = {body.name: body.mesh.nb_faces for body in bodies}
+        if bemio_import:
+            LOG.warning('Bemio data does not include mesh data. mesh=True is ignored.')
+        else:
+            # TODO: Store full mesh...
+            bodies = list({result.body for result in results})  # Filter out duplicate bodies in the list of results
+            nb_faces = {body.name: body.mesh.nb_faces for body in bodies}
 
-        def name_or_str(c):
-            return c.name if hasattr(c, 'name') else str(c)
-        quad_methods = {body.name: name_or_str(body.mesh.quadrature_method) for body in bodies}
+            def name_or_str(c):
+                return c.name if hasattr(c, 'name') else str(c)
+            quad_methods = {body.name: name_or_str(body.mesh.quadrature_method) for body in bodies}
 
-        if len(nb_faces) > 1:
-            dataset.coords['nb_faces'] = ('body_name', [nb_faces[name] for name in dataset.coords['body_name'].data])
-            dataset.coords['quadrature_method'] = ('body_name', [quad_methods[name] for name in dataset.coords['body_name'].data])
-        else:
-            def the_only(d):
-                """Return the only element of a 1-element dictionnary"""
-                return next(iter(d.values()))
-            dataset.coords['nb_faces'] = the_only(nb_faces)
-            dataset.coords['quadrature_method'] = the_only(quad_methods)
+            if len(nb_faces) > 1:
+                dataset.coords['nb_faces'] = ('body_name', [nb_faces[name] for name in dataset.coords['body_name'].data])
+                dataset.coords['quadrature_method'] = ('body_name', [quad_methods[name] for name in dataset.coords['body_name'].data])
+            else:
+                def the_only(d):
+                    """Return the only element of a 1-element dictionnary"""
+                    return next(iter(d.values()))
+                dataset.coords['nb_faces'] = the_only(nb_faces)
+                dataset.coords['quadrature_method'] = the_only(quad_methods)
 
     # HYDROSTATICS
     if hydrostatics:
-        bodies = list({result.body for result in results})
-        dataset = xr.merge([dataset, hydrostatics_dataset(bodies)])
+        if bemio_import:
+            LOG.warning('Bemio data import being used, hydrostatics=True is ignored.')
+        else:
+            bodies = list({result.body for result in results})
+            dataset = xr.merge([dataset, hydrostatics_dataset(bodies)])
 
     dataset.attrs.update(attrs)
     dataset.attrs['capytaine_version'] = __version__
     return dataset
 
 
+#######################
+#  Import from Bemio  #
+#######################
+
+def dataframe_from_bemio(bemio_obj, wavenumber, wavelength):
```

Comment:
I suggest we move this function into a separate file `capytaine/io/bemio.py` or something like that.

Reply:
Done in 231bf6d.